sections/futures/TradeCrossMargin/DepositWithdrawCrossMargin.tsx

```diff
@@ -0,0 +1,289 @@
+
+			setTxState('submitting');
+			const tx = await crossMarginAccountContract.deposit(wei(amount).toBN());
+			monitorTransaction({
+				txHash: tx.hash,
+				onTxConfirmed: () => {
+					setTxState('complete');
+					handleRefetch('account-margin-change');
+					onComplete?.();
+					onDismiss();
+				},
+			});
+		} catch (err) {
+			setError(err.message);
+			setTxState('none');
+			logError(err);
+		}
+	}, [
+		crossMarginAccountContract,
+		amount,
+		monitorTransaction,
+		handleRefetch,
+		onComplete,
+		onDismiss,
+	]);
+
+	const depositMargin = useCallback(async () => {
+		try {
+			const wallet = await signer?.getAddress();
+
+			if (!crossMarginAccountContract || !wallet) throw new Error('No cross margin account');
+			const weiAmount = wei(amount ?? 0, 18);
+			const allowance = await susdContract?.allowance(wallet, crossMarginAccountContract.address);
+
+			if (wei(allowance).lt(weiAmount)) {
+				setTxState('approving');
+				const tx = await susdContract?.approve(
+					crossMarginAccountContract.address,
+					constants.MaxUint256
+				);
+				if (tx?.hash) {
+					monitorTransaction({
+						txHash: tx.hash,
+						onTxConfirmed: () => {
+							submitDeposit();
+						},
+					});
+				}
+			} else {
+				submitDeposit();
+			}
+		} catch (err) {
+			setError(err.message);
+			setTxState('none');
+			logError(err);
+		}
+	}, [crossMarginAccountContract, amount, signer, susdContract, monitorTransaction, submitDeposit]);
+
+	const withdrawMargin = useCallback(async () => {
+		try {
+			if (!crossMarginAccountContract) throw new Error('No cross-margin account');
+			setTxState('submitting');
+			const tx = await crossMarginAccountContract.withdraw(wei(amount).toBN());
+			monitorTransaction({
+				txHash: tx.hash,
+				onTxConfirmed: () => {
+					setTxState('complete');
+					handleRefetch('account-margin-change');
+					onComplete?.();
+					onDismiss();
+				},
+			});
+		} catch (err) {
+			setError(err.message);
+			setTxState('none');
+			logError(err);
+		}
+	}, [
+		crossMarginAccountContract,
+		amount,
+		monitorTransaction,
+		handleRefetch,
+		onComplete,
+		onDismiss,
+	]);
+
+	// TODO: Get tx fee
+	const transactionFee = wei('0.1');
+
+	const disabledReason = useMemo(() => {
+		const amtWei = wei(amount || 0);
+		if (transferType === 0) {
+			const total = wei(freeMargin).add(amtWei);
+			if (total.lt(MIN_DEPOSIT_AMOUNT))
+				return t('futures.market.trade.margin.modal.deposit.min-deposit');
+			if (amtWei.gt(susdBal)) return t('futures.market.trade.margin.modal.deposit.exceeds-balance');
+		} else {
+			if (amtWei.gt(freeMargin))
+				return t('futures.market.trade.margin.modal.deposit.exceeds-balance');
+		}
+	}, [amount, freeMargin, transferType, susdBal, t]);
+
+	const handleSetMax = React.useCallback(() => {
+		setAmount(susdBal.toString());
+	}, [susdBal]);
+
+	const onChangeTab = (selection: number) => {
+		setTransferType(selection);
+		setAmount('');
+	};
+
+	return (
+		<StyledBaseModal
+			title={t(
```

Comment:
Should this title be centered?
<img width="430" alt="image" src="https://user-images.githubusercontent.com/10401554/188013407-fbb0bcf7-32b7-426e-ac9d-d378ea29a074.png">

Reply:
Incorrect, checked the designs